### PR TITLE
Feature/nex 875/create helpers

### PR DESCRIPTION
**Pull request by bartlomiejmarszal**
Created: 
- Encoder.php
- Normalizer.php
- ProctorioConfig.php
- SignatureBuilder.php

https://oat-sa.atlassian.net/browse/NEX-877

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -19,7 +19,6 @@
     "ext-curl": "*"
   },
   "require-dev": {
-    "phpunit/phpunit": "^8.5"
```
Comment (mike-ionut-mihai-sandu-tao) [inline composer.json]: why ?

File: src/ProctorioConfig.php [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+ */
+
+namespace oat\Proctorio;
+
+use Ramsey\Uuid\Uuid;
+
+class ProctorioConfig
+{
+    public const PROCTORIO_URL = 'https://%s' . '15499ws.proctor.io/6521ca945bd84cfc85d2767da06aa7c8';
+    // https://{account region}5499ws.proctor.io/6521ca945bd84cfc85d2767da06aa7c8
```
Comment (mike-ionut-mihai-sandu-tao) [inline src/ProctorioConfig.php]: let remove this comments,  let put this inside the doc that i requested

File: tests/unit/EncoderTest.php [hunk 3]
```diff
@@ -0,0 +1,58 @@
+<?php declare(strict_types=1);
```
Comment (mike-ionut-mihai-sandu-tao) [inline tests/unit/EncoderTest.php]: if you are doing this here, please do it everywhere

File: tests/unit/EncoderTest.php [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+
+    /**
+     * @dataProvider getUrlSources
+     */
+    public function testEncode(string $stringToEncode, string $expected): void
+    {
+        $result = $this->subject->encode($stringToEncode);
+        $this->assertEquals($expected, $result);
+    }
+}
```
Comment (mike-ionut-mihai-sandu-tao) [inline tests/unit/EncoderTest.php]: please run the phpcs --standard=PSR12 on this

File: composer.json (hunk 1 above)
Comment (bartlomiejmarszal) [reply composer.json]: mistake. Fixed

File: tests/unit/EncoderTest.php (hunk 3 above)
Comment (bartlomiejmarszal) [reply tests/unit/EncoderTest.php]: fixed

File: src/ProctorioConfig.php (hunk 2 above)
Comment (bartlomiejmarszal) [reply src/ProctorioConfig.php]: I moved all coments to doc file.

File: tests/unit/EncoderTest.php (hunk 4 above)
Comment (bartlomiejmarszal) [reply tests/unit/EncoderTest.php]: I don't know but my sniffer is not recognizing missing empty lines in tests files

**Comment by mike-ionut-mihai-sandu-tao** [comment]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

File: src/ProctorioConfig.php [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+ * Copyright (c) 2020 (original work) Open Assessment Technologies SA (under the project TAO-PRODUCT);
+ */
+
+namespace oat\Proctorio;
+
+use Ramsey\Uuid\Uuid;
+
+class ProctorioConfig
+{
+    public const PROCTORIO_URL = 'https://%s' . '15499ws.proctor.io/6521ca945bd84cfc85d2767da06aa7c8';
```
Comment (boajer) [inline src/ProctorioConfig.php:27]: Do we really need this part `6521ca945bd84cfc85d2767da06aa7c8` hardcoded?

Comment (mike-ionut-mihai-sandu-tao) [reply src/ProctorioConfig.php:27]: that's the URL from the docs, and i think they have that has declared as an endpoint
normally within the service we can change but this is according to the docs

**Comment by boajer** [comment]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [ ] New code is respecting code style rules
- [ ] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [ ] Acceptance criteria are respected
- [ ] Pull request title and description are meaningful

**Pull request merged**